scheduler/scheduler_sync.daemon

```diff
@@ -60,25 +61,40 @@ readonly SITE_ID="${SITE_NUM[$RADARID]}"    # Get site id corresponding to corre
 readonly MAXWELL="dataman@maxwell.usask.ca"
 readonly DEST="radar@localhost"
 readonly DEST_PORT="5${SITE_ID}522"
-# readonly DEST_DIR="/home/radar/borealis_schedules"
-DEST_DIR="/home/radar/test_scheduler_sync/"     # Testing
+readonly DEST_DIR="/home/radar/borealis_schedules"
 
 ###################################################################################################
 
 exec &>> $LOGFILE # Redirect STDOUT and STDERR to $LOGFILE
 printf "Starting $0 $RADARID at $(date --utc "+%Y%m%d %H:%M:%S UTC")\n"
 
-inotifywait --monitor --syslog --timefmt '%Y%m%d.%H:%M:%S' --format '%T %w %e %f' -e modify -e attrib $WATCH_DIR | \
+# Inotify will trigger on all modify (file content change) and attrib (file metadata change) events.
+# These events happen in close succession, leading to many triggers at once. To reduce the number of
+# times rsync is executed, a check is done to ensure its been at least $filter_time seconds since
+# the last time the schedule was synced. This will effectively filter out all the redundant inotify
+# triggers and ensure rsync only executes when there's actually changes to synchronize.
+filter_time=1
+endtime=0
+
+inotifywait --monitor --timefmt '%Y%m%d.%H:%M:%S' --format '%T %w %e %f' -e modify,attrib $WATCH_DIR | \
 while read TIME DIRECTORY ACTION FILE # These must line up with format parameters
 do
-    if [[ "$FILE" =~ .*$RADARID\.scd$ ]]   # Only interested in .scd files for the specified site
+    # The conditional filters the inotify output to only trigger the syncronization when the .scd
+    # file for the site specified by RADARID is modified.
+    if [[ "$FILE" =~ .*$RADARID\.scd$ ]]
     then
-        printf "$(date --utc "+%Y%m%d %H:%M:%S UTC") - Schedule modified for $RADARID\n"
+        # Only execute the code once every $filter_time seconds
+        cur_time=$(date "+%s")
+        if [[ $(($endtime + $filter_time)) -lt $cur_time ]]; then
```

Comment:
Shouldn't this be: if endtime+filtertime -gt cur_time?
then you'll have to initialize endtime to date +%s as well.

Currently I think it will always sleep 1 second for long waits between runs, but never for a short wait between runs (except the first time)

Reply:
I copied the implementation from [borealis.daemon](https://github.com/SuperDARNCanada/data_flow/blob/main/inotify_daemons/borealis.daemon) in data_flow, so I think the `-lt` should work. 

I think you're right that there is a problem though, as right now I'm `sleep`ing for 1s within the loop, and my filter time is also set to 1s so I may not be filtering anything. I think changing `filter_time` to 2 should fix this though?

Reply:
Looking again, I think my current implementation should work? The conditional is saying if less than 1 second has passed since the conditional last finished execution, the conditional won't execute. The script should execute correctly for all runs greater than 1 s apart, while for any shorter waits I don't want it to execute

Reply:
My idea for the `sleep` was to wait a second after the first inotify event triggers so that any changes to the file will be 100% finished. If there isn't a sleep there I think there could be the chance that the rsync executes before all the changes to the schedule file are complete. Then, the changes that occurred after rsync executed wouldn't be synced to site until the next time the schedule is changed

Reply:
That makes sense, thanks for the explanation.